### PR TITLE
[Technical] Add tests for header and footer

### PR DESCRIPTION
**Pull request by dasdom**
This commit adds tests for header and footer in DynamicTableViewController.
There are still some tests to write for this class.

## Checklist

* [x] Test your changes as thoroughly as possible before you commit them. Preferably, automate your test by unit/integration tests.
* [x] If this PR comes from a fork, please [Allow edits from maintainers](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/allowing-changes-to-a-pull-request-branch-created-from-a-fork)
* [x] Set a speaking title. Format: {task_name} (closes #{issue_number}). For example: Use logger (closes # 41)
* [x] [Link your Pull Request to an issue](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) (if applicable)
* [x] Create Work In Progress [WIP] pull requests only if you need clarification or an explicit review before you can continue your work item.
* [x] Make sure that your PR is not introducing _unnecessary_ reformatting (e.g., introduced by on-save hooks in your IDE)

## Description

Improve test coverage in DynamicTableViewController.

### CODE REVIEW

**Comment by dasdom** [comment]
Hm... archiving fails. I don't really understand why. Will look into it tomorrow night.

File: src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewControllerFooterTests.swift
Comment (ChristianKienle) [inline src/xcode/ENA/ENA/Source/Scenes/DynamicTableViewController/__tests__/DynamicTableViewControllerFooterTests.swift:37]:
```suggestion
				XCTFail("Unable to instantiate DynamicTableViewController from DynamicTableViewControllerFake.storyboard")
```

**Review by ChristianKienle — APPROVED** [review]
Sorry for taking a bit longer this time. I have approved the changes but suggested a small change. If you get to it you may consider just accepting the suggestion.

Otherwise let's wait for the second approval.

Thanks again for adding more tests! <3

**Comment by dasdom** [comment]
It didn't take longer. I just managed a few minutes ago to make SwiftLint happy.

**Review by inf2381 — APPROVED** [review]

**Pull request merged**
